### DISCUSSION TITLE
Can a MenuItem with children be clickable?

### DISCUSSION THREAD

**Discussion by IncPlusPlus** [AvaloniaUI/Avalonia#8871] [Q&A]
I noticed that if a `MenuItem` has any `Items` set on it, it no longer will execute the command it is bound to when clicked. I know this is sorta the expected behavior but in my app I'd want this particular group of `MenuItem` to be clickable regardless of if they have child `Items`. I stepped through the code with a debugger and found that click events [check whether the item has a submenu](https://github.com/AvaloniaUI/Avalonia/blob/c7c6450004138ca80f1ee668131a83a334726e98/src/Avalonia.Controls/Platform/DefaultMenuInteractionHandler.cs#L435) before raising the click event. There's also [a similar check](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Platform/DefaultMenuInteractionHandler.cs#L156) for the key down event too.

I've got a context menu that looks like the following.

![image](https://user-images.githubusercontent.com/6992149/187711570-7e4851e1-087d-4150-8d6e-ee44bcd1a048.png)

I was hoping to be able to disable the submenu checks for all `MenuItem`s that are children of the "Set Category" `MenuItem` so that I could let the user select any category they like instead of just the narrowest one. It seems like this is pretty core behavior so I don't have high hopes but would it be possible to modify this behavior?

**Comment by timunie** [comment]
Maybe you can try to listen to PointerPressedEvent instead.  

Also maybe a Command will fire, I have not tried it, but may worth a try. 

Finally you still can tunnel events before it is handeld by the control.
